token/program-2022/tests/default_account_state.rs

```diff
@@ -0,0 +1,232 @@
+#![cfg(feature = "test-bpf")]
+
+mod program_test;
+use {
+    program_test::{TestContext, TokenContext},
+    solana_program_test::tokio,
+    solana_sdk::{
+        instruction::InstructionError, program_option::COption, pubkey::Pubkey, signature::Signer,
+        signer::keypair::Keypair, transaction::TransactionError, transport::TransportError,
+    },
+    spl_token_2022::{
+        error::TokenError, extension::default_account_state::DefaultAccountState,
+        instruction::AuthorityType, state::AccountState,
+    },
+    spl_token_client::token::{ExtensionInitializationParams, TokenError as TokenClientError},
+    std::convert::TryFrom,
+};
+
+#[tokio::test]
+async fn success_init_default_acct_state_frozen() {
+    let default_account_state = AccountState::Frozen;
+    let mut context = TestContext::new().await;
+    context
+        .init_token_with_mint(vec![ExtensionInitializationParams::DefaultAccountState {
+            state: default_account_state,
+        }])
+        .await
+        .unwrap();
+    let TokenContext {
+        decimals,
+        mint_authority,
+        token,
+        ..
+    } = context.token_context.unwrap();
+
+    let state = token.get_mint_info().await.unwrap();
+    assert_eq!(state.base.decimals, decimals);
+    assert_eq!(
+        state.base.mint_authority,
+        COption::Some(mint_authority.pubkey())
+    );
+    assert_eq!(state.base.supply, 0);
+    assert!(state.base.is_initialized);
+    assert_eq!(state.base.freeze_authority, COption::None);
```

Comment:
This is really specific, but what do you think about adding in a check to make sure that this situation doesn't happen?  Ie, initializing a mint that has default account state of Frozen, but then no configured freeze authority.  It's either that, or use include a configuration authority in the default account state extension

Reply:
Ah, good point. I'll add that check to InitializeMint (has to happen there, since that ix is last).